### PR TITLE
Implement CTLine and CTRunGetImageBounds

### LINKED ISSUE #1981: CoreText: implement CTLineGetImageBounds and CTRunGetImageBounds

### PR DESCRIPTION
**Pull request by aballway**
Fixes #1981

### CODE REVIEW

File: Frameworks/include/CppUtils.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ std::basic_ostream<T>& operator<<(std::basic_ostream<T>& os, const CFRange& rang
 
 #pragma region CGPoint
 inline bool operator==(const CGPoint& lhs, const CGPoint& rhs) {
-    return ((std::abs(lhs.x - rhs.x) < 0.00001) && (std::abs(lhs.y - rhs.y) < 0.00001));
+    return lhs.x == rhs.x && lhs.y == rhs.y;
```
Comment (DHowett-MSFT) [inline Frameworks/include/CppUtils.h]: This will almost certainly make tests fail on ARM. We had these set up for a reason. What's up?

Comment (DHowett-MSFT) [inline Frameworks/include/CppUtils.h]: Oh, I see. Interesting.

Comment (aballway) [inline Frameworks/include/CppUtils.h]: Tests passed on ARM

**Comment by aballway** [comment]
Leading, which is some added space above a line typesetters use for multiline text, is not part of the image bounds.
There was a bug in CTRunGetTypographicBounds for ranges with location > 0

**Pull request merged**